lib/dal/src/management/mod.rs

```diff
@@ -721,61 +724,78 @@ impl<'a> ManagementOperator<'a> {
         })
     }
 
-    async fn prepare_for_connection(
+    async fn input_socket_id(
         &mut self,
-        source_component_id: ComponentId,
-        connection: &ManagementConnection,
-    ) -> ManagementResult<(ComponentId, OutputSocketId, ComponentId, InputSocketId)> {
-        let from_variant_id = self
+        component_id: ComponentId,
+        socket: &str,
+    ) -> ManagementResult<InputSocketId> {
+        let variant_id = self
             .component_schema_map
-            .variant_for_component_id(self.ctx, source_component_id)
+            .variant_for_component_id(self.ctx, component_id)
             .await?;
-
         // if the map was already constructed this does nothing
         self.socket_map
-            .add_sockets_for_variant(self.ctx, from_variant_id)
+            .add_sockets_for_variant(self.ctx, variant_id)
             .await?;
+        let socket_id = self.socket_map.input_socket_id(variant_id, socket).ok_or(
+            ManagementError::InputSocketDoesNotExist(component_id, socket.to_owned()),
+        )?;
+        Ok(socket_id)
+    }
 
-        let destination_component_id = self.get_real_component_id(&connection.to.component)?;
-
-        let to_variant_id = self
+    async fn output_socket_id(
+        &mut self,
+        component_id: ComponentId,
+        socket: &str,
+    ) -> ManagementResult<OutputSocketId> {
+        let variant_id = self
             .component_schema_map
-            .variant_for_component_id(self.ctx, destination_component_id)
+            .variant_for_component_id(self.ctx, component_id)
             .await?;
-
+        // if the map was already constructed this does nothing
         self.socket_map
-            .add_sockets_for_variant(self.ctx, to_variant_id)
+            .add_sockets_for_variant(self.ctx, variant_id)
             .await?;
+        let socket_id = self.socket_map.output_socket_id(variant_id, socket).ok_or(
+            ManagementError::OutputSocketDoesNotExist(component_id, socket.to_owned()),
+        )?;
+        Ok(socket_id)
+    }
 
-        let source_output_socket_id = self
-            .socket_map
-            .output_socket_id(from_variant_id, &connection.from)
-            .ok_or(ManagementError::OutputSocketDoesNotExist(
-                source_component_id,
-                destination_component_id,
-                connection.from.to_owned(),
-            ))?;
-
-        let destination_input_socket_id = self
-            .socket_map
-            .input_socket_id(to_variant_id, &connection.to.socket)
-            .ok_or(ManagementError::OutputSocketDoesNotExist(
-                source_component_id,
-                destination_component_id,
-                connection.to.socket.to_owned(),
-            ))?;
+    async fn prepare_for_connection(
```

Comment:
prepare_for_connection now supports both input and output directions, so this change lets it figure out which side the `component_id` goes on ("from" for output, "to" for input).